kornia/contrib/extract_patches.py

```diff
@@ -1,10 +1,12 @@
-from typing import Optional, Tuple, Union
+from typing import Optional, Tuple, Union, cast
 
 import torch
 import torch.nn as nn
 import torch.nn.functional as F
 from torch.nn.modules.utils import _pair
 
+PadType = Union[Tuple[int, int], Tuple[int, int, int, int]]
```

Comment:
just one side note here -- torchscript might not work well here for the `Union` type

Reply:
Does this extend to function args? If not, I could just replace `PadType` with `Union[Tuple[int, int], Tuple[int, int, int, int]]`.

Reply:
Seems Union support will be added in Pytorch 1.11.0. Ref: pytorch/pytorch#64234

Reply:
I think we can address the torchscript issue in a later PR. This PR is mainly to unify APIs and ensure correct behaviour. From some initial tests, solving scripting issues would involve removing `Union` (which would reduce flexibility of the functions) and some other functions like `torch.is_tensor` which needs some discussion.

Reply:
agree -- torchscript it's another discussion that has been bothering us for a while :)

Are you in our slack ? We can have an open discussion there and collect feedback from out developers community.

Reply:
Sounds good. Recently joined the kornia slack so we can continue there